Added functionality to add metadata to PDFDocument

LukasLoeffler opened the pull request:
While using this library I noticed that there is no possibility to add/change the metadata of the PDF-Document like title, author or producer. This pull request will add this feature to the library. I am not entirely sure about the conventions of the type declarations, so it would be great if anyone had a look and provide feedback. 

- Added function setMetadata which expects object of type Metadata
- Function setMetadata sets metadata for producer, author, title, creator
- The function ensures the document is present by internal _ensureDoc function
- Due to the _ensureDoc check the funtion is async
- Added type declarations for Metadata by adding an interface

mojoaxel reviewed (changes requested):
This looks really good! A very nice new feature! 🎉 

@LukasLoeffler Could you maybe add some usage info to the README and maybe a test or two!?

Let me know if I can help!

LukasLoeffler commented:
I updated the README the following way:
- Added the setMetadata function to the exported functions.
- Added a code example

The tests were extended by a test which sets the metadata with the new method. Afterwards pdf-lib is used to load the output file and check if the set and read metadata match.

I don't think a test which compares two files via pdfDiff is not necessary in consideration of how pdfDiff checks the files. Two files with different metadata will come out as equal from the pdfDiff perspective.

mojoaxel commented:
@LukasLoeffler Thanks again!

LukasLoeffler commented:
I have to thank for the great package 👍

mojoaxel commented:
:rocket: Released as [v4.2.0](https://github.com/nbesli/pdf-merger-js/releases/tag/4.2.0)
